core/rag/simple_rag.py

```
@@ -102,18 +102,70 @@ def _score(query: str, text: str) -> float:
     """Return top-k relevant chunks concatenated with brief headers."""
     items = _get_index()
     if not items:
         return ""
-    scored = sorted(
-        ((src, ch, _score(query, ch)) for src, ch in items), key=lambda x: x[2], reverse=True
-    )
+
+    # Enhanced scoring with multiple factors
+    scored = []
+    query_lower = query.lower()
+    query_words = set(_tokenize(query))
```

Flake8 (7.3.0)

[error] local variable 'query_words' is assigned to but never used

(F841)